src/go/rpk/pkg/cli/redpanda/mode.go

```diff
@@ -63,3 +63,27 @@ func executeMode(fs afero.Fs, p *config.Params, mode string) error {
 	}
 	return cfg.SetMode(fs, mode)
 }
+
+const modeHelpText = `Enable a default configuration mode
+
+This command allows you to set one of the following modes: Development,
+Production, or Recovery.
+
+PRODUCTION
+
+  - Enforces optimal runtime checks.
+  - Enables all rpk tuners, for you to run 'rpk redpanda tune'.
+
+DEVELOPMENT
+
+  - Disables optimal checks and bypasses fsync, which results in unrealistically
```

Comment:
Disabling fsync is _only_ done in mode=dev-container, not standard development mode.

Reply:
But developer mode opts into dev-container at the moment when running `rpk redpanda start` https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/redpanda/start.go#L159-L161

The only way to opt-out of that will be starting with developer mode and `--mode ""` because we only have one mode at the moment.

Reply:
Wow, I forgot we did that. Alright then, my comment can be discarded :D